src/app/pages/ordenesdepago/components/set-datoscompromiso/set-datoscompromiso.component.ts

```diff
@@ -117,15 +118,17 @@ export class SetDatoscompromisoComponent implements OnInit, OnDestroy {
     });
 
     this.datosCompromiso.get('actaRecibido').valueChanges.subscribe(valor => {
-      this.store.dispatch(getEntradaAlmacen({query: {EstadoMovimientoId__Nombre__startswith: 'Entrada', Detalle__json_contains: `{"acta_recibido_id":${valor}}`}}));
+      if (this.flagActa) {
+        this.store.dispatch(getEntradaAlmacen({query: {EstadoMovimientoId__Nombre__startswith: 'Entrada',
+        Detalle__json_contains: `{"acta_recibido_id":${valor}}`}}));
+      }
       this.subEntradaAlmacen$ = this.store.select(selectEntradaAlmacen).subscribe((action) => {
-        if (action && action.EntradaAlmacen[0]) {
+        if (action && action.EntradaAlmacen && action.EntradaAlmacen[0]) {
```

Comment:
Quizás lo siguiente sea suficiente ... ?
```suggestion
        if (action && action.EntradaAlmacen && action.EntradaAlmacen.length) {
```